### PR TITLE
Vector DB CDK: Add "from field" embedding strategy

### LINKED ISSUE #29711: Vector db CDK: Add "from field" embedding strategy
In this strategy, the user has to specify a field name as a string. The field has to be an array of numbers (otherwise the sync fails) that contains the vector representation of the record already. Then, the vector field is removed from the document metadata.

### PR DESCRIPTION
**Pull request by flash1293**
## What

Fixes https://github.com/airbytehq/airbyte/issues/29711 by adding a new `Embedder` class and the matching config model.

## How

This PR adds a new embedding strategy "from field". To support this case and the "calculate embeddings downstream" case (like Weaviate vectorizers: https://weaviate.io/developers/weaviate/modules/retriever-vectorizer-modules ), it makes the following changes:
* Move embedding calculation to the writer level
* Pass in embeddings to the "index" method of the indexer
* Change embedder interface to take the chunks, not the texts

## 🚨 User Impact 🚨

As the interfaces for Writer and Indexer change, all connectors using the vector db CDK needs to be adapted to the new structure.

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/README.md [hunk 1]
```diff
@@ -24,15 +24,12 @@ This is how the components interact:
...
-└┬────────┘└───────┘└─────────────────┘
-┌▽───────┐                             
-│Embedder│                             
-└────────┘                             
+┌▽───────────────────────────────┐                
+│Writer                          │                
+└┬─────────┬──────────┬────────┬─┘                
+┌▽───────┐┌▽────────┐┌▽──────┐┌▽────────────────┐
+│Embedder││MyIndexer││Batcher││DocumentProcessor│
+└────────┘└─────────┘└───────┘└─────────────────┘
```
Comment (pedroslopez) [inline airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/README.md:32]: Fancy!

File: airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/embedder.py [hunk 2]
```diff
@@ -79,8 +90,8 @@ def check(self) -> Optional[str]:
             return format_exception(e)
         return None
 
-    def embed_texts(self, texts: List[str]) -> List[List[float]]:
```
Comment (pedroslopez) [inline airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/embedder.py:82]: Should this be renamed to `embed_chunks`?

File: airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/embedder.py [hunk 3]
```diff
@@ -100,10 +111,54 @@ def check(self) -> Optional[str]:
...
+        Check that the field exists, is a list of numbers and is the correct size. If not, raise an AirbyteTracedException explaining the problem.
+        """
+        embeddings = []
+        for chunk in chunks:
+            data = chunk.record.data
+            if self.config.field_name not in data:
+                raise AirbyteTracedException(
+                    internal_message="Embedding vector field not found",
+                    failure_type=FailureType.config_error,
+                    message=f"Record {str(data)[:250]}...  does not contain embedding vector field {self.config.field_name}. Please check your embedding configuration, the embedding vector field has to be set correctly on every record.",
```
Comment (pedroslopez) [inline airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/embedder.py:143]: Nice error messages! is it worth saying which stream this record is in?

**Review by pedroslopez — APPROVED** [review]
Nice!

File: airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/embedder.py (hunk 2 above)
Comment (flash1293) [reply airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/embedder.py:82]: yes, updated

**Pull request merged**